src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/StaticsInfoHashtableNode.cs

```diff
@@ -39,6 +39,41 @@ public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         public override ObjectNodeSection Section => _externalReferences.Section;
         public override bool StaticDependenciesAreComputed => true;
         protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
+        
+
+        /// <summary>
+        /// Helper method to compute the dependencies that would be needed by a hashtable entry for statics info lookup.
+        /// This helper is used by EETypeNode, which is used by the dependency analysis to compute the statics hashtable
+        /// entries for the compiled types.
+        /// </summary>
+        public static DependencyList GetStaticsInfoDependencies(NodeFactory factory, TypeDesc type)
+        {
+            DependencyList dependencies = new DependencyList();
+
+            if (type is MetadataType && type.HasInstantiation && !type.IsCanonicalSubtype(CanonicalFormKind.Any))
+            {
+                MetadataType metadataType = (MetadataType)type;
+
+                // NOTE: The StaticsInfoHashtable entries need to reference the gc and non-gc static nodes through an indirection cell.
+                // The StaticsInfoHashtable entries only exist for static fields on generic types.
+
+                if (metadataType.GCStaticFieldSize.AsInt > 0)
+                {
+                    dependencies.Add(factory.Indirection(factory.TypeGCStaticsSymbol(metadataType)), "GC statics indirection for StaticsInfoHashtable");
+                }
+
+                if (metadataType.NonGCStaticFieldSize.AsInt > 0 || factory.TypeSystemContext.HasLazyStaticConstructor(type))
```

Comment:
Probably not right now. I initially thought I needed this check adding in other places (a bunch of the template layout code) but I don't - it handles class constructors and non-GC statics with separate template layout entries. If we have to write this check a second time then I would switch to a yes.